frontend/src/components/Workspace/Visualize/Plot/ImagePlot.tsx

```diff
@@ -554,36 +703,69 @@ const ImagePlotChart = memo(function ImagePlotChart({
   }
 
   const renderActionRoi = () => {
-    if (!roiDataState?.length || outputKey !== "cell_roi") return null
-    if (!isAddRoi) {
-      return <LinkDiv onClick={addRoi}>Add ROI</LinkDiv>
+    if (!roiFilePath || !roiFilePath.includes(CELL_ROI)) return null
+    if (action) {
+      return (
+        <>
+          {action !== ADD_ROI ? (
+            <BoxDiv>
+              ROI Selecteds: [{pointClick.map((item) => item.z).join(",")}]
+            </BoxDiv>
+          ) : null}
+          <BoxDiv sx={{ display: "flex", gap: 1 }}>
+            <LinkDiv
+              sx={{
+                color:
+                  action === DELETE_ROI
+                    ? "#F84E1B"
+                    : action === MERGE_ROI
+                    ? "#6619A9"
+                    : "default",
+                display: "flex",
+                gap: 1,
+                textDecoration: "none",
+                cursor: "default",
+              }}
+            >
+              {`[${action}]`}
+            </LinkDiv>
+            <LinkDiv onClick={onCancelAdd}>Cancel</LinkDiv>
```

Comment:
@sanglevinh 
Currently, the Cancel link in ROI Add/Merge/Delete does not seem to be working. Please check and adjust.